mavros_extras/src/plugins/trajectory.cpp

```diff
@@ -0,0 +1,579 @@
+
+			auto enu_velocity_point_5 = ftf::transform_frame_ned_enu(Eigen::Vector3d(trajectory.point_5[3], trajectory.point_5[4], trajectory.point_5[5]));
+			trajectory_desired->point_5.velocity.x = enu_velocity_point_5.x();
+			trajectory_desired->point_5.velocity.y = enu_velocity_point_5.y();
+			trajectory_desired->point_5.velocity.z = enu_velocity_point_5.z();
+			auto enu_acceleration_point_5 = ftf::transform_frame_ned_enu(Eigen::Vector3d(trajectory.point_5[6], trajectory.point_5[7], trajectory.point_5[8]));
+			trajectory_desired->point_5.acceleration_or_force.x = enu_acceleration_point_5.x();
+			trajectory_desired->point_5.acceleration_or_force.y = enu_acceleration_point_5.y();
+			trajectory_desired->point_5.acceleration_or_force.z = enu_acceleration_point_5.z();
+			double yaw_enu_point_5 = wrap_pi((M_PI / 2.0f) - trajectory.point_5[9]);
+			trajectory_desired->point_5.yaw = yaw_enu_point_5;
+			trajectory_desired->point_5.yaw_rate = trajectory.point_5[10];
+
+			// [[[end]]] (checksum: 71b88583d4e270dfe47465a17af7099f)
+		} else {
+			// [[[cog:
+			// for p in "12345":
+			//     cog.outl("trajectory_desired->point_{p}.velocity.x = trajectory.point_{p}[3];".format(**locals()))
+			//     cog.outl("trajectory_desired->point_{p}.yaw = wrap_pi((M_PI / 2.0f) - trajectory.point_{p}[4]);".format(**locals()))
+			//     cog.outl("trajectory_desired->point_{p}.yaw_rate = trajectory.point_{p}[5]; \n".format(**locals()))
+			// ]]]
+			trajectory_desired->point_1.velocity.x = trajectory.point_1[3];
+			trajectory_desired->point_1.yaw = wrap_pi((M_PI / 2.0f) - trajectory.point_1[4]);
+			trajectory_desired->point_1.yaw_rate = trajectory.point_1[5];
+
+			trajectory_desired->point_2.velocity.x = trajectory.point_2[3];
+			trajectory_desired->point_2.yaw = wrap_pi((M_PI / 2.0f) - trajectory.point_2[4]);
+			trajectory_desired->point_2.yaw_rate = trajectory.point_2[5];
+
+			trajectory_desired->point_3.velocity.x = trajectory.point_3[3];
+			trajectory_desired->point_3.yaw = wrap_pi((M_PI / 2.0f) - trajectory.point_3[4]);
+			trajectory_desired->point_3.yaw_rate = trajectory.point_3[5];
+
+			trajectory_desired->point_4.velocity.x = trajectory.point_4[3];
+			trajectory_desired->point_4.yaw = wrap_pi((M_PI / 2.0f) - trajectory.point_4[4]);
+			trajectory_desired->point_4.yaw_rate = trajectory.point_4[5];
+
+			trajectory_desired->point_5.velocity.x = trajectory.point_5[3];
+			trajectory_desired->point_5.yaw = wrap_pi((M_PI / 2.0f) - trajectory.point_5[4]);
+			trajectory_desired->point_5.yaw_rate = trajectory.point_5[5];
+
+			// [[[end]]] (checksum: 64b954c5837b398a1de95fa281977885)
+		}
+
+		std::copy(trajectory.point_valid.begin(), trajectory.point_valid.end(), trajectory_desired->point_valid.begin());
+
+		trajectory_desired_pub.publish(trajectory_desired);
+	}
+
+	float wrap_pi(float a)
+	{
+		if (!std::isfinite(a)) {
+			return a;
+		}
+
+		while (a >= M_PI) {
+			a -= (M_PI * 2.0f);
+		}
+
+		while (a < -M_PI) {
+			a += (M_PI * 2.0f);
+		}
+
```

Comment:
I would propose the following implementation, using modulo; ensuring constant execution time.

```c++
return fmod(a + M_PI, 2.0*M_PI) - M_PI;
```